crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -694,7 +694,9 @@ where
                 &last_archived_block_encoded,
                 block_object_mappings,
             )
-            .expect("Incorrect parameters for archiver");
+            .unwrap_or_else(|error| {
+                panic!("Incorrect parameters for archiver: {error:?} {last_segment_header:?}")
+            });
```

Comment:
This is fine, but I suspect one of the historical reasons it was panicking is that function didn't return an error, but now it does, so might be a good thing to return it from here as well (in a simpler form, while using `error!()` macro to print `last_segment_header`).